integreat_cms/cms/views/pages/page_form_view.py

```diff
@@ -359,33 +359,39 @@ def post(
                     slug=user_slug,
                     language=language,
                 ).first()
-                other_translation_link = reverse(
-                    "page_versions",
-                    kwargs={
-                        "page_id": other_translation.page.id,
-                        "language_slug": language.slug,
-                        "region_slug": region.slug,
-                        "selected_version": other_translation.version,
-                    },
-                )
-                message = _(
-                    "The slug was changed from '{user_slug}' to '{slug}', "
-                    "because '{user_slug}' is already used by <a>{translation}</a>.",
-                ).format(
-                    user_slug=user_slug,
-                    slug=page_translation_form.cleaned_data["slug"],
-                    translation=other_translation,
-                )
-                messages.warning(
-                    request,
-                    translate_link(
-                        message,
-                        attributes={
-                            "href": other_translation_link,
-                            "class": "underline hover:no-underline",
+                if other_translation:
+                    other_translation_link = reverse(
+                        "page_versions",
+                        kwargs={
+                            "page_id": other_translation.page.id,
+                            "language_slug": language.slug,
+                            "region_slug": region.slug,
+                            "selected_version": other_translation.version,
                         },
-                    ),
-                )
+                    )
+                    message = _(
+                        "The slug was changed from '{user_slug}' to '{slug}', "
+                        "because '{user_slug}' is already used by <a>{translation}</a>.",
+                    ).format(
+                        user_slug=user_slug,
+                        slug=page_translation_form.cleaned_data["slug"],
+                        translation=other_translation,
+                    )
+                    messages.warning(
+                        request,
+                        translate_link(
+                            message,
+                            attributes={
+                                "href": other_translation_link,
+                                "class": "underline hover:no-underline",
+                            },
+                        ),
+                    )
+                else:
+                    logger.error(
+                        "Renaming slug failed, because translation is not existant"
+                    )
+                    messages.error("This didn't work. Please try again in a moment.")
```

Comment:
I don't think this is the correct message to give. *(I tried to be concise, but it turned into a bit of a rant… sorry)*
We **already saved the page translation form** on line 351, which was apparently successful or otherwise we would have aborted with an exception. Afterwards we just **checked whether the cleaned slug differed** from the one originally submitted by the browser. In the `clean_…()` method for that field we employed [`generate_unique_slug()` from `slug_utils.py`](https://github.com/digitalfabrik/integreat-cms/blob/dc5f02332bffbfcabf3a04ded93a662fa3ee9fec/integreat_cms/cms/utils/slug_utils.py#L89-L181), which is the part of the code that is trying to avoid collisions *(I didn't read through this, I just wanted to confirm the clean method is where this happens)*.

Let me reiterate, interestingly, this check happens after the translation has already been saved – now we're just looking to generate meaningful user facing messages. Since we only know that the slug changed, we **assume** there must have been a collision. It would be nice to tell the user which page we collided with, so we try to query it. Our error happens if, for whatever reason, we don't actually find a translation with that original slug. And this is the case that we want to convey at this point:

Not that saving it failed, or changing the slug was unsuccessful, we just didn't find a translation holding the originally requested slug.
So what should we do? We could either try to understand which other reasons there could be for `generate_unique_slug()` to come up with a slug different from the requested one and make appropriate explanation messages for that. Or we could just say that we don't know why, but the slug got changed and display that as a warning, not an error, affirming that otherwise, everything has been saved and no data is lost.

Reply:
I think it's weird to tell the user "hey something happened but we don't know why our system did that" right in the message automatically given to them. But we could at least tell them it was saved with a different slug, not citing any reason. Maybe the user doesn't care about the slug, but better make it explicit.

I guess we could adapt the original message and just not mention a reason. As for the logging message, maybe something like this?
```suggestion
                    logger.warning(
                        "Slug was changed from the one the user provided, but we can't find the translation that already used it: %s (cleaned to %s)",
                        user_slug,
                        page_translation_form.cleaned_data["slug"],
                    )
                    messages.warning(
                        _("The slug was changed from '{user_slug}' to '{slug}.").format(
                            user_slug=user_slug,
                            slug=page_translation_form.cleaned_data["slug"],
                        )
                    )
```